File: leet_code/binary_search/b_search.js
const bSearch = (arr, target) => {
  if (arr.length < 1 || (arr.length === 1 && arr[0] !== target)) return -1;

  let left = 0;
  let right = arr.length - 1;

  while (left <= right) {
    let mid = Math.floor((left + right) / 2);
    if (arr[mid] === target) return mid;
    arr[mid] < target ? left = mid + 1 : right = mid - 1;
  }
  return -1;
};

File: leet_code/binary_search/b_search.rb
def b_search(arr, target)
    return -1 if arr.length < 1 || (arr.length == 1 && arr[0] != target)

    left = 0
    right = arr.length - 1
    mid = (left + right) / 2

    while left <= right
        mid = (left + right) / 2
        return mid if arr[mid] == target
        arr[mid] < target ? left = mid + 1 : right = mid - 1
    end
    -1
end
